### PR TITLE
Add "tanzu context update ucp-active-resource" command to set UCP active resource

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR adds the "tanzu context update ucp-active-resource" command to set the UCP active resource of a give UCP context

**Changes summary:**
- Add a hidden command "tanzu context update ucp-active-resource" to set the UCP active resource \(org/project/space) for the given UCP context. This command would also update the kubeconfig referenced by the UCP context to point to the updated active resource.
- It would return error if the context is not UCP context type
- It would return error if user tries to set the space without providing the project name. 
- The command would not check for validity of the project and space names.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Created a UCP context and later set the ucp-active-resource to space1. The tanzu config file and kubeconfig file are updated successfully
```
❯ ./bin/tanzu context create myucp --endpoint https://ucp-aria.stacks.bluesky.tmc-dev.cloud.vmware.com --type application-engine  --staging
[i] API token env var is set

[ok] successfully created a Application Engine(UCP) context


//kubeconfig file after context created (the server URL would be referring to the org
apiVersion: v1
clusters:
- cluster:
    server: https://ucp-aria.stacks.bluesky.tmc-dev.cloud.vmware.com/org/ee04bfae-a665-4f20-a5b9-d8b043180252
  name: tanzu-cli-myucp/current
contexts:
- context:
    cluster: tanzu-cli-myucp/current
    user: tanzu-cli-myucp-user
  name: tanzu-cli-myucp
current-context: tanzu-cli-myucp
kind: Config
preferences: {}
users:
- name: tanzu-cli-myucp-user
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1
      args:
      - context
      - get-token
      - myucp
      command: tanzu
      env: []
      interactiveMode: Never
      provideClusterInfo: false


// Now set the ucp active context to space-test which is a child of project-test
❯ ./bin/tanzu context update ucp-active-resource myucp --project project-test --space space-test

//kubeconfig after setting the ucp active context to project-test/space-test
apiVersion: v1
clusters:
- cluster:
    server: https://ucp-aria.stacks.bluesky.tmc-dev.cloud.vmware.com/org/ee04bfae-a665-4f20-a5b9-d8b043180252/project/project-test/space/space-test
  name: tanzu-cli-myucp/current
contexts:
- context:
    cluster: tanzu-cli-myucp/current
    user: tanzu-cli-myucp-user
  name: tanzu-cli-myucp
current-context:  tanzu-cli-myucp
kind: Config
preferences: {}
users:
- name: tanzu-cli-myucp-user
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1
      args:
      - context
      - get-token
      - myucp
      command: tanzu
      env: []
      interactiveMode: Never
      provideClusterInfo: false
```

-Tested setting the active ucp resource to project and verified the context additionalMetadata is updated successfully and kubeconfig is also updated successfully

--Tested setting the active ucp resource back to org  and verified the context additionalMetadata is updated successfully and kubeconfig is also updated successfully

### Release note

```release-note

```

### CODE REVIEW

File: pkg/command/context_test.go [hunk 1]
```diff
@@ -331,6 +336,110 @@ clusterOpts:
...
+			ucpContext.Target = configtypes.TargetK8s
+			err = config.SetContext(ucpContext, false)
+			Expect(err).To(BeNil())
+
+			err = setUCPCtxActiveResource(cmd, []string{testContextName})
+			Expect(err).ToNot(BeNil())
+			Expect(err.Error()).To(ContainSubstring(`context "test-context" is not of type UCP`))
+
+		})
+		It("should return error if user tries to set space as active resource with out providing project name", func() {
```
Comment (vuil) [inline pkg/command/context_test.go:397]: nit: without

File: pkg/command/context_test.go [hunk 2]
```diff
@@ -331,6 +336,110 @@ clusterOpts:
...
+		cmd := &cobra.Command{}
+
+		BeforeEach(func() {
+			testKubeconfigFilePath := "../fakes/config/kubeconfig1.yaml"
+			kubeconfigFilePath, err = os.CreateTemp("", "config")
+			Expect(err).To(BeNil())
+			err = copy.Copy(testKubeconfigFilePath, kubeconfigFilePath.Name())
+			Expect(err).To(BeNil())
+
+			os.Setenv("KUBECONFIG", kubeconfigFilePath.Name())
```
Comment (vuil) [inline pkg/command/context_test.go:354]: q: wonder if this is necessary

File: pkg/fakes/config/kubeconfig1.yaml [hunk 3]
```diff
@@ -12,6 +12,10 @@ clusters:
       insecure-skip-tls-verify: true
       server: https://foo.org:443
     name: foo-cluster
+  - cluster:
+      insecure-skip-tls-verify: true
```
Comment (vuil) [inline pkg/fakes/config/kubeconfig1.yaml:16]: nit: would removing this line impact testing the functionality? If not I'd take it out.

**Review by vuil — COMMENTED** [review]
lgtm. thanks!
a few minor questions and typo.
Also, I suggest mentioning in the PR that this command also does not check the validity  of the project/space.

**Review by vuil — APPROVED** [review]

File: pkg/command/context_test.go (hunk 1 above)
Comment (prkalle) [reply pkg/command/context_test.go]: updated

File: pkg/command/context_test.go (hunk 2 above)
Comment (prkalle) [reply pkg/command/context_test.go]: Not needed. Updated.

File: pkg/fakes/config/kubeconfig1.yaml (hunk 3 above)
Comment (prkalle) [reply pkg/fakes/config/kubeconfig1.yaml]: It shouldn't affect the the testing functionality. Removed it. Thanks

**Pull request merged**